src/wp-includes/i18n/class-wp-translation-file.php

```diff
@@ -0,0 +1,302 @@
+				return false;
+		}
+	}
+
+	/**
+	 * Returns all headers.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array<string, string> Headers.
+	 */
+	public function headers() {
+		if ( ! $this->parsed ) {
+			$this->parse_file();
+		}
+		return $this->headers;
+	}
+
+	/**
+	 * Returns all entries.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array<string, string[]> Entries.
+	 */
+	public function entries() {
+		if ( ! $this->parsed ) {
+			$this->parse_file();
+		}
+
+		return $this->entries;
+	}
+
+	/**
+	 * Returns the current error information.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return bool|string Error
+	 */
+	public function error() {
+		return $this->error;
+	}
+
+	/**
+	 * Returns the file name.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return string File name.
+	 */
+	public function get_file() {
+		return $this->file;
+	}
+
+	/**
+	 * Translates a given string.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $text String to translate.
+	 * @return false|string Translation(s) on success, false otherwise.
+	 */
+	public function translate( string $text ) {
+		if ( ! $this->parsed ) {
+			$this->parse_file();
+		}
+
+		return $this->entries[ $text ] ?? false;
+	}
+
+	/**
+	 * Returns the plural form for a count.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param int $number Count.
+	 * @return int Plural form.
+	 */
+	public function get_plural_form( int $number ) {
+		if ( ! $this->parsed ) {
+			$this->parse_file();
+		}
+
+		// In case a plural form is specified as a header, but no function included, build one.
+		if ( null === $this->plural_forms && isset( $this->headers['plural-forms'] ) ) {
+			$this->plural_forms = $this->make_plural_form_function( $this->headers['plural-forms'] );
+		}
+
+		if ( is_callable( $this->plural_forms ) ) {
+			/**
+			 * Plural form.
+			 *
+			 * @var int $result Plural form.
+			 */
+			$result = call_user_func( $this->plural_forms, $number );
+			return $result;
+		}
+
+		// Default plural form matches English, only "One" is considered singular.
+		return ( 1 === $number ? 0 : 1 );
+	}
+
+	/**
+	 * Makes a function, which will return the right translation index, according to the
+	 * plural forms header.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $expression Plural form expression.
+	 * @return callable(int $num): int Plural forms function.
+	 */
+	public function make_plural_form_function( string $expression ) {
+		try {
+			$handler = new Plural_Forms( rtrim( $expression, ';' ) );
+			return array( $handler, 'get' );
+		} catch ( Exception $e ) {
+			// Fall back to default plural-form function.
+			return $this->make_plural_form_function( 'n != 1' );
+		}
+	}
+
+	/**
+	 * Creates a new WP_Translation_File instance for a given file.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $file     Source file name.
+	 * @param string $filetype Desired target file type.
+	 * @return string|false Transformed translation file contents on success, false otherwise.
+	 */
+	public static function transform( string $file, string $filetype ) {
```

Comment:
Nit-pick: Maybe colocate all the static methods either at the top or bottom of the class? I think it makes it a bit more intuitive to scan the class and see which static methods are present, as their responsibility is quite different from the instance methods.